cast/src/main.rs

```diff
@@ -53,13 +56,36 @@ enum Commands {
 #[tokio::main]
 async fn main() -> Result<()> {
     let cli = Cli::parse();
-
-    // todo: #2052 take network from scarb config if flag not provided
-    let network_name = cli.network.unwrap_or_else(|| {
+    let mut maybe_network_name = None;
+    if let Some(network_name) = cli.network {
+        maybe_network_name = Some(network_name);
+    } else {
+        let metadata = scarb_metadata::MetadataCommand::new()
+            .inherit_stderr()
+            .current_dir(current_dir().expect("failed to read current directory"))
+            .no_deps()
+            .exec()
+            .unwrap();
+        if metadata.packages.len() != 1 {
+            anyhow::bail!("invalid number of packages obtained from scarb metadata");
+        }
+        let package = &metadata.packages[0];
+        if let Some(tool) = &package.manifest_metadata.tool {
+            if let Some(protostar_tool) = tool.get("protostar") {
+                if let Some(network) = protostar_tool.get("network") {
+                    if let Some(network_str) = network.as_str() {
+                        maybe_network_name = Some(network_str.to_string());
+                    }
+                }
+            }
+        }
+    }
+    if maybe_network_name.is_none() {
         // todo: #2107
         eprintln!("{}", style("No --network flag passed!").red());
         std::process::exit(1);
```

Comment:
Consider moving to type system
https://github.com/software-mansion/scarb/blob/6484cf7b60e166475becc175e6a66e6d95dff6e8/scarb/src/bin/scarb/errors.rs#L7

Reply:
1. this is not a change introduced in this PR, but ok we can change this.
2. what do u mean? you want to raise a scarb error instead of returning process code 1 or something else? i didnt understand.

Reply:
Irrelevant as per #2113, I will remove this altogether under this task, but thanks for the tip Maciek, I will remember that! :)

Reply:
I guess it would be good to wrap the errors in a custom error that makes the program exit with non-zero exit code in the future